pkg/network/dhcp/masquerade_test.go

```diff
@@ -85,8 +79,20 @@ var _ = Describe("Masquerade DHCP configurator", func() {
 			return expectedConfig
 		}
 
-		generateExpectedConfigIPv6Enabled := func(vmiSpecNetwork *v1.Network, macString *string, mtu int, ifaceName string, subdomain string) cache.DHCPConfig {
-			expectedConfig := generateExpectedConfigIPv6Disabled(vmiSpecNetwork, macString, mtu, ifaceName, subdomain)
+		generateExpectedConfigOnlyIPv4Enabled := func(vmiSpecNetwork *v1.Network, macString *string, mtu int, ifaceName string, subdomain string) cache.DHCPConfig {
+			expectedConfig := generateExpectedConfig(vmiSpecNetwork, macString, mtu, ifaceName, subdomain)
+			ipv4, _ := netlink.ParseAddr(expectedIpv4)
+			ipv4Gateway, _ := netlink.ParseAddr(expectedIpv4Gateway)
```

Comment:
I do not think we need to protect against bad test writing in the same manner we do for the tested code.
If someone wrote the test with bad input, we could panic or do nothing like here.

In this specific case, if we do nothing, it will blow up as a panic anyway, as you cannot use `*` on a `nil` (this is what you will get if there is an error).